crypto/keys/keybase_base.go

```diff
@@ -34,6 +46,30 @@ type (
 	}
 )
 
+// WithKeygenFunc applies an overridden key generation function to generate the private key.
+func WithKeygenFunc(f PrivKeyGenFunc) KeybaseOption {
```

Comment:
How is this API supposed to be used? I can't find any example in this PR

Reply:
I'll just copy the usage in Ethermint:
```
	return clientkeys.NewKeyringFromHomeFlag(buf, keys.WithKeygenFunc(ethermintKeygenFunc))
```

would you like me to document this somewhere specifically?

Reply:
I see now. I think it'd best to drop a code example in a test file, see https://golang.org/pkg/testing/#hdr-Examples